src/movies.js

```diff
@@ -1,16 +1,199 @@
-// Iteration 1: All directors? - Get the array of all directors.
+// let trials = [
+//     {
+//       "title": "The Shawshank Redemption",
+//       "year": 1994,
+//       "director": "Frank Darabont",
+//       "duration": "2h 22min",
+//       "genre": [
+//         "Crime",
+//         "Drama"
+//       ],
+//       "rate": 9.3
+//     },
+//     {
+//       "title": "The Godfather",
+//       "year": 1972,
+//       "director": "Francis Ford Coppola",
+//       "duration": "2h 55min",
+//       "genre": [
+//         "Crime",
+//       ],
+//       "rate": 9.2
+//     },
+//     {
+//       "title": "The Godfather: Part II",
+//       "year": 1974,
+//       "director": "Francis Ford Coppola",
+//       "duration": "3h",
+//       "genre": [
+//         "Crime",
+//         "Drama"
+//       ],
+//       "rate": 9
+//     },  {
+//         "title": "The Godfather: Part dsII",
+//         "year": 1974,
+//         "director": "Francis Ford Coppola",
+//         "duration": "3h",
+//         "genre": [
+//           "Crime",
+//           "Drama"
+//         ],
+//         "rate": 4
+//       },
+//     {
+//       "title": "The Dark Knight",
+//       "year": 2008,
+//       "director": "Christopher Nolan",
+//       "duration": "2h 32min",
+//       "genre": [
+//         "Action",
+//         "Crime",
+//         "Drama",
+//         "Thriller"
+//       ],
+//       "rate": 9
+//     }]
+    
+    // Iteration 1: All directors? - Get the array of all directors.
+ 
+
+// console.log(getAllDirectors(trials));
+function getAllDirectors(movies) {
+    const unfilteredDirs = movies.map(function(movie) {
+        return movie.director;
+    });
+    const filteredDirs = unfilteredDirs.filter(function (director, index) {
+        if (unfilteredDirs.indexOf(director, index + 1) === -1) return true;
```

Comment:
1. You can simply return the condition and leave the if else statement and the returning true and false out.
2. Are you uniqueifying the array? Could there be a more minimal solution to do this? I believe Jan discussed "Set" in his lecture.